runtime/compiler/x/codegen/J9TreeEvaluator.cpp

```diff
@@ -7762,11 +7762,13 @@ J9::X86::TreeEvaluator::VMnewEvaluator(
    startLabel->setStartInternalControlFlow();
    fallThru->setEndInternalControlFlow();
 
+   bool enableTLHBatchClearing = fej9->tlhHasBeenCleared();
+
 #ifdef J9VM_GC_NON_ZERO_TLH
    // If we can skip zero init, and it is not outlined new, we use the new TLH
    // same logic also appears later, but we need to do this before generate the helper call
    //
-   if (node->canSkipZeroInitialization() && !comp->getOption(TR_DisableDualTLH) && !comp->getOptions()->realTimeGC())
+   if (node->canSkipZeroInitialization() && (enableTLHBatchClearing || !comp->getOption(TR_DisableDualTLH)) && !comp->getOptions()->realTimeGC())
```

Comment:
Why does the code guarded by this condition only handle `new` and `newarray` ? For example, would we allocate `anewarray` and `multianewarray` differently somehow ?

Reply:
multianewarray initialization doesn't reach here, but anewarray does.  I'm not sure why it isn't included here (it never was based on the long history of this code), but Z and P both support it.  I can add a case for that.

Reply:
Fixed in [6ddd240](https://github.com/eclipse-openj9/openj9/pull/19514/commits/6ddd240f09a35ae6a35893d66a4bad32a9ab133c)

Reply:
Okay thanks. 

Should the commit message say "Do not perform zero initialization of TLH objects when batch clearing enabled" ?